datastore.go

```diff
@@ -101,7 +116,11 @@ func (d *datastore) QueryNew(q dsq.Query) (dsq.Results, error) {
 	prefix := []byte(q.Prefix)
 	opt := badger.DefaultIteratorOptions
 	opt.PrefetchValues = !q.KeysOnly
-	it := d.DB.NewIterator(opt)
+
+	txn := d.DB.NewTransaction(false)
+	defer txn.Discard()
```

Comment:
Can you discard a transaction before you close the iterator?

Reply:
I have no idea why it worked, good catch.